Glyssen/Rules/CharacterGroupGenerator.cs

```diff
@@ -914,13 +1021,118 @@ public static void DistributeBooksAmongNarratorGroups(List<AuthorStats> authorSt
 				}
 			}
 
-			private void AssignExtraBiblicalCharactersToExtraBiblicalGroups(List<CharacterDetail> includedCharacterDetails)
+			private void AssignExtraBiblicalCharactersToExtraBiblicalGroups(List<CharacterDetail> includedCharacterDetails,
+				ProjectDramatizationPreferences dramatizationPreferences)
+			{
+				var narratorPrefix = CharacterVerseData.GetCharacterPrefix(CharacterVerseData.StandardCharacter.Narrator);
+
+				// get standard characters not yet assigned
+				foreach (var character in includedCharacterDetails.Where(c => !Groups.Any(g => g.CharacterIds.Contains(c.CharacterId))).ToList())
+				{
+					var characterType = character.StandardCharacterType;
+					switch (characterType)
+					{
+						case CharacterVerseData.StandardCharacter.BookOrChapter:
+							AddExtraBiblicalToGroupOrNarratorGroup(character.CharacterId, BookTitleChapterGroup, characterType,
+								narratorPrefix, dramatizationPreferences);
+							break;
+
+						case CharacterVerseData.StandardCharacter.ExtraBiblical:
+							AddExtraBiblicalToGroupOrNarratorGroup(character.CharacterId, SectionHeadGroup, characterType,
+								narratorPrefix, dramatizationPreferences);
+							break;
+
+						case CharacterVerseData.StandardCharacter.Intro:
+							AddExtraBiblicalToGroupOrNarratorGroup(character.CharacterId, BookIntroductionGroup, characterType,
+								narratorPrefix, dramatizationPreferences);
+							break;
+					}
+				}
+			}
+
+			private void AddExtraBiblicalToGroupOrNarratorGroup(string characterId, CharacterGroup group,
+				CharacterVerseData.StandardCharacter characterType, string narratorPrefix, ProjectDramatizationPreferences dramatizationPreferences)
 			{
-				if (ExtraBiblicalGroup == null)
+				if (group != null)
```

Comment:
Might want to just put this up in the previous method (after taking my other suggestion there) to avoid the cost of a method call and pushing a bunch of parameters that are not needed in this case.